frontend/ui/elements/refresh_menu_table.lua

```diff
@@ -1,30 +1,51 @@
+local Device = require("device")
 local UIManager = require("ui/uimanager")
-local util = require("ffi/util")
 local _ = require("gettext")
+local Screen = Device.screen
+local T = require("ffi/util").template
 
-local function custom_1() return G_reader_settings:readSetting("refresh_rate_1") or 12 end
-
-local function custom_2() return G_reader_settings:readSetting("refresh_rate_2") or 22 end
-
-local function custom_3() return G_reader_settings:readSetting("refresh_rate_3") or 99 end
+local function custom(refresh_rate_type)
+    local default_value
+    if refresh_rate_type == "refresh_rate_1" then
+        default_value = 12
+    elseif refresh_rate_type == "refresh_rate_2" then
+        default_value = 22
+    else
+        default_value = 99
+    end
+    return G_reader_settings:readSetting(refresh_rate_type) or default_value
+end
 
-local function custom_input(name)
-    return {
-        title = _("Input page number for a full refresh"),
-        type = "number",
-        hint = "(1 - 99)",
-        callback = function(input)
-            local rate = tonumber(input)
-            G_reader_settings:saveSetting(name, rate)
-            UIManager:setRefreshRate(rate)
-        end,
+local function spinWidgetSetRefresh(touchmenu_instance, refresh_rate_type)
+    local SpinWidget = require("ui/widget/spinwidget")
+    local items = SpinWidget:new{
+        width = Screen:getWidth() * 0.6,
+        value = custom(refresh_rate_type),
+        value_min = 0,
+        value_max = 200,
+        value_step = 1,
+        value_hold_step = 10,
+        ok_text = _("Set refresh"),
+        title_text = _("Set custom refresh rate"),
+        callback = function(spin)
+            G_reader_settings:saveSetting(refresh_rate_type, spin.value)
+            UIManager:setRefreshRate(spin.value)
+            touchmenu_instance:updateItems()
+        end
     }
+    UIManager:show(items)
 end
 
+
 return {
     text = _("Full refresh rate"),
     separator = true,
     sub_item_table = {
+        {
+            text = _("Never refresh"),
```

Comment:
Or just `Never` or `Manually` ? @Frenzie ?

Reply:
`Manual and UI`?

Reply:
I use `Never`.